src/queryResult/queryResultWebViewController.ts

```diff
@@ -93,7 +93,14 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
                     message.numberOfRows,
                 );
             let currentState = this.getQueryResultState(message.uri);
-            if (currentState.isExecutionPlan) {
+            if (
+                currentState.isExecutionPlan &&
+                // check if the current result set is the result set that contains the xml plan
+                currentState.resultSetSummaries[message.batchId][
+                    message.resultId
+                ].columnInfo[0].columnName ===
+                    "Microsoft SQL Server 2005 XML Showplan"
+            ) {
```

Comment:
If Karl's comment below ("is this always true?") means checking each summary, this may be a good thing to extract out to a utility function

Reply:
i'm not sure if it makes sense to me to extract the new implementation of it into a utility function. in my head, this was already a utility function. Let me know though if you feel that I should still make a separate utility function with the new implementation!